### PR TITLE
[TCFMM-15] Fix "invert prices" TODO

### PR DESCRIPTION
**Pull request by Martoon-00**
[//]: # (This is a template of a pull request template.)
[//]: # (You should modify it considering specifics of a particular repository and put it there.)
[//]: # (Comments like this are meta-comments, they shouldn't be present in the final template.)
[//]: # (Comments starting with '

Resolves TCFMM-15

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

[//]: # (Update link to style guide if necesary or remove if it's not present)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: ligo/swaps.mligo [hunk 1]
```diff
@@ -91,10 +92,11 @@ let rec y_to_x_rec (p : y_to_x_rec_param) : y_to_x_rec_param =
...
             let next_tick = get_tick p.s.ticks i_u in
             let sqrt_price_new = next_tick.sqrt_price in
-            (* How much dx will we receive for going all the wax to lo. *)
 
-            (* FIXME this is wrong, invert prices etc *)
-            let dx = Bitwise.shift_right (p.s.liquidity * (assert_nat (sqrt_price_new - p.s.sqrt_price))) 90n in
+            (* How much dx will we receive for going all the way to lo. *)
+            (* From 6.16 formula: dx = L * (1 / old sqrt_price - 1 / new sqrt_price), where dx is how X decreases *)
+            let dx = ceildiv (p.s.liquidity * Bitwise.shift_left (assert_nat (sqrt_price_new - p.s.sqrt_price)) 90n)
+                             (sqrt_price_new * p.s.sqrt_price) in
```
Comment (dcastro) [inline ligo/swaps.mligo:99]: I think we also have to shift `sqrt_price_new * p.s.sqrt_price` right by 180.

Comment (Martoon-00) [reply ligo/swaps.mligo:99]: If we decrease the divisible by `2^90` times and the divisor by `2^180` times, that would be the same as increasing the divisible by `2^90` times (modulo potential issues with precision loss) - does it make sense?

Or you have something different in mind? Could you write down the formula then?

Comment (dcastro) [reply ligo/swaps.mligo:99]: Oh yeah, good point! :smile:

**Review by dcastro — APPROVED** [review]

**Comment by dcastro** [comment]
@rinn7e can we get your review on this please?

**Review by rinn7e — APPROVED** [review]
Looks good for me.

**Pull request merged**